azure/sentinel/main.tf

```diff
@@ -0,0 +1,88 @@
+##########################
+# Log Analytics Workspace
+##########################
+
+resource "azurerm_log_analytics_workspace" "main" {
+  count               = var.log_analytics_workspace_name != null ? 1 : 0
+  name                = var.log_analytics_workspace_name
+  location            = var.location
+  tags                = var.tags
+  resource_group_name = var.resource_group_name
+  retention_in_days   = var.log_analytics_workspace_retention
+  sku                 = var.log_analytics_workspace_sku
```

Comment:
Add a precondition here to prevent creation of `azurerm_log_analytics_workspace` if `log_analytics_workspace_id` is specified.

If we don't add validation, the module could create a workspace whilst simultaneously not using the workspace (based on the logic of the local). Therefore we need to validate that "if `var.log_analytics_workspace_id` is set, then `var.log_analytics_workspace_name` should be not set and vice versa.

An easy way to do this is to create a condition where the length of both variables cannot both be equal to 0.

EG:

```
lifecycle {
    postcondition {
      condition = (length(var.log_analytics_workspace_id) == 0) != (length(var.log_analytics_workspace_name) == 0)`
      }
  }
```

This will ensure that one of them is set but not both and not neither.

Reply:
precondition added